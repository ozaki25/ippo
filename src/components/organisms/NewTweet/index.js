import React from 'react';
import { Classes, Dialog } from '@blueprintjs/core';
import propTypes from 'prop-types';
import TweetForm from 'src/components/organisms/TweetForm';

const styles = {
  dialog: {
    margin: '0 15px',
  },
};

const NewTweet = ({ createTweet, isOpen, onClose, hashtag, refetch, authUser }) => {
  const onFinished = () => {
    refetch();
    onClose();
  };
  return (
    <Dialog
      title="ツイート"
      isOpen={isOpen}
      onClose={onClose}
      style={styles.dialog}
      isCloseButtonShown
    >
      <div className={Classes.DIALOG_BODY}>
        <TweetForm
          createTweet={createTweet}
          hashtag={hashtag}
          onFinished={onFinished}
          authUser={authUser}
        />
      </div>
    </Dialog>
  );
};

NewTweet.displayName = 'NewTweet';

NewTweet.propTypes = {
  createTweet: propTypes.func.isRequired,
  isOpen: propTypes.bool,
  onClose: propTypes.func,
  hashtag: propTypes.string.isRequired,
  refetch: propTypes.func.isRequired,
  authUser: propTypes.shape({
    uid: propTypes.string.isRequired,
    displayName: propTypes.string.isRequired,
  }).isRequired,
};

NewTweet.defaultProps = {
  isOpen: false,
  onClose: null,
};

export default NewTweet;
