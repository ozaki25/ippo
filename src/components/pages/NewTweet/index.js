import React from 'react';
import { TextField, Typography } from '@material-ui/core';
import styled from 'styled-components';
import propTypes from 'prop-types';
import RoundedButton from 'src/components/atoms/RoundedButton';
import IconImage from 'src/components/atoms/IconImage';
import Container from 'src/components/templates/Container';

const inputProps = {
  style: {
    paddingTop: '10px',
  },
};

const Wrapper = styled.div`
  margin-left: 58px;
`;

const StyledIconImage = styled(IconImage)`
  margin-left: -58px;
`;

const Buttons = styled.div`
  margin: 8px 0;
  text-align: right;
`;

class NewTweet extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tweet: `
#${this.props.hashtag}`,
      error: false,
    };
    this.tweet = React.createRef();
  }

  componentDidMount() {
    this.tweet.current.setSelectionRange(0, 0);
  }

  onChange = event => this.setState({ [event.target.name]: event.target.value });

  onClickUpload = () => console.log('TODO');

  onClickTweet = async () => {
    const { hashtag, createTweet, authUser, history } = this.props;
    const tweet = {
      text: this.state.tweet,
      hashtag,
      name: authUser.displayName,
      uid: authUser.uid,
      time: new Date().toString(),
    };
    try {
      const result = await createTweet({
        variables: { tweet },
      });
      console.log(result);
      history.goBack('refetch');
    } catch (e) {
      this.setState({ error: e.toString() });
      console.log(e);
    }
  };

  render() {
    const { authUser, history, firebase } = this.props;
    const { tweet, error } = this.state;
    const disabled = !tweet.trim();
    return (
      <Container title="ツイート" back authUser={authUser} history={history} firebase={firebase}>
        <Wrapper>
          <StyledIconImage src="/icon.png" />
          {error && <Typography color="error">{error}</Typography>}
          <TextField
            name="tweet"
            rows="4"
            margin="none"
            placeholder="いまどうしてる？"
            InputProps={inputProps}
            inputRef={this.tweet}
            onChange={this.onChange}
            value={this.state.tweet}
            variant="filled"
            multiline
            fullWidth
            autoFocus
          />
          <Buttons>
            <RoundedButton color="primary" variant="outlined" onClick={this.onClickUpload}>
              アップロード
            </RoundedButton>
            <RoundedButton color="primary" disabled={disabled} onClick={this.onClickTweet}>
              ツイート
            </RoundedButton>
          </Buttons>
        </Wrapper>
      </Container>
    );
  }
}

NewTweet.displayName = 'NewTweet';

NewTweet.propTypes = {
  hashtag: propTypes.string.isRequired,
  createTweet: propTypes.func.isRequired,
  authUser: propTypes.shape({
    displayName: propTypes.string.isRequired,
    uid: propTypes.string.isRequired,
  }).isRequired,
  history: propTypes.shape({
    push: propTypes.func.isRequired,
    goBack: propTypes.func.isRequired,
  }).isRequired,
  firebase: propTypes.object.isRequired,
};

NewTweet.defaultProps = {};

export default NewTweet;
