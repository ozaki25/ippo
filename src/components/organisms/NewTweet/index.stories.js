import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import NewTweet from '.';

const stories = storiesOf('organisms/NewTweet', module);

const props = {};

stories.add('通常パターン', () => <NewTweet {...props} />);
