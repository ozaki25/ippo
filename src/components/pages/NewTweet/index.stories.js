import React from 'react';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import NewTweet from '.';

export default {
  title: 'pages/NewTweet',
};

const props = {
  hashtag: text('hashtag', 'test_event'),
  createTweet: action('create'),
  authUser: {
    uid: '123',
    displayName: 'テストユーザ',
  },
  history: {
    push: action('push'),
    goBack: action('goBack'),
    location: {},
  },
  firebase: {},
  parentTweet: {
    tweet: {},
  },
};

const defaultTweetProps = {
  history: {
    push: action('push'),
    goBack: action('goBack'),
    location: {
      search: '?tweet=デフォルト',
    },
  },
};

const joinProps = {
  history: {
    push: action('push'),
    goBack: action('goBack'),
    location: {
      search: '?tweet=参加します！&type=join',
    },
  },
};

const leaveProps = {
  history: {
    push: action('push'),
    goBack: action('goBack'),
    location: {
      search: '?tweet=キャンセルします&type=leave',
    },
  },
};

const parentTweetProps = {
  parentTweet: {
    tweet: {
      id: '123',
      name: '親',
      text: '親ツイートです',
      time: '2019/1/22',
      hashtag: 'test',
    },
  },
};

export const 通常パターン = () => <NewTweet {...props} />;

export const エラー = () => (
  <NewTweet
    {...props}
    createTweet={() =>
      new Promise((resolve, reject) =>
        reject(new Error('Network error: Failed to fetch')),
      )
    }
  />
);

export const デフォルトツイート = () => (
  <NewTweet {...props} {...defaultTweetProps} />
);

export const 参加 = () => <NewTweet {...props} {...joinProps} />;

export const キャンセル = () => <NewTweet {...props} {...leaveProps} />;

export const 親ツイートあり = () => (
  <NewTweet {...props} {...parentTweetProps} />
);
