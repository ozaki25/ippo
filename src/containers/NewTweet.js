import { graphql } from 'react-apollo';
import { compose } from 'recompose';
import { withRouter } from 'react-router-dom';
import { withFirebase } from 'context/firebase';
import query from 'graphql/query';
import mutation from 'graphql/mutation';
import NewTweet from 'components/pages/NewTweet';
import { withAuthorization } from 'hoc/Sessions';
import withHashtag from 'hoc/withHashtag';
import withParentTweet from 'hoc/withParentTweet';

export default compose(
  withAuthorization,
  withRouter,
  withFirebase,
  withHashtag,
  withParentTweet,
  graphql(mutation.createTweet, { name: 'createTweet' }),
  graphql(query.tweet, {
    name: 'parentTweet',
    options: ({ match: { params }, hashtag, parentId }) => ({
      variables: { hashtag, id: parentId },
    }),
  }),
)(NewTweet);
