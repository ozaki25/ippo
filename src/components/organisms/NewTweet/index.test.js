import React from 'react';
import { snapshot } from 'test/helpers';
import NewTweet from '.';

const props = {};

snapshot('NewTweet/nomal', <NewTweet {...props} />);
