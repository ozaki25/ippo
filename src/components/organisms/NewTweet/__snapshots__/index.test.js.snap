// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: NewTweet/nomal 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <NewTweet
    isOpen={false}
    onClose={null}
  >
    <Blueprint3.Dialog
      canOutsideClickClose={true}
      isCloseButtonShown={true}
      isOpen={false}
      onClose={null}
      style={
        Object {
          "margin": "0 15px",
        }
      }
      title="ツイート"
    >
      <Blueprint3.Overlay
        autoFocus={true}
        backdropProps={Object {}}
        canEscapeKeyClose={true}
        canOutsideClickClose={true}
        className="bp3-overlay-scroll-container"
        enforceFocus={true}
        hasBackdrop={true}
        isCloseButtonShown={true}
        isOpen={false}
        lazy={true}
        onClose={null}
        style={
          Object {
            "margin": "0 15px",
          }
        }
        title="ツイート"
        transitionDuration={300}
        transitionName="bp3-overlay"
        usePortal={true}
      />
    </Blueprint3.Dialog>
  </NewTweet>
</MockRouter>
`;
